app/controllers/events/view/sessions/list.js

```diff
@@ -1,204 +1,269 @@
-      title        : 'Submission Date'
-    },
-    {
-      propertyName : 'lastModifiedAt',
-      template     : 'components/ui-table/cell/cell-simple-date',
-      dateFormat   : 'MMMM DD, YYYY - HH:mm A',
-      title        : 'Last Modified'
-    },
-    {
-      propertyName   : 'is-mail-sent',
-      title          : 'Email Sent',
-      template       : 'components/ui-table/cell/events/view/sessions/cell-is-mail-sent',
-      disableSorting : true
-    },
-    {
-      template         : 'components/ui-table/cell/events/view/sessions/cell-buttons',
-      title            : 'Actions',
-      disableSorting   : true,
-      disableFiltering : true
-    },
-    {
-      template         : 'components/ui-table/cell/events/view/sessions/cell-lock-session',
-      title            : 'Lock Session',
-      disableSorting   : true,
-      disableFiltering : true
+export default class extends Controller.extend(EmberTableControllerMixin) {
+  @mapBy('model.feedbacks', 'session.id') ratedSessions;
+
+  @computed()
+  get columns() {
+    return [
+      {
+        name            : 'State',
+        valuePath       : 'status',
+        isSortable      : true,
+        headerComponent : 'tables/headers/sort',
+        cellComponent   : 'ui-table/cell/events/view/sessions/cell-session-state'
+      },
+      {
+        name            : 'Title',
+        valuePath       : 'title',
+        extraValuePaths : ['event', 'isLocked'],
+        isSortable      : true,
+        headerComponent : 'tables/headers/sort',
+        cellComponent   : 'ui-table/cell/events/view/sessions/cell-session-title',
+        width           : 160,
+        actions         : {
+          viewSession   : this.viewSession.bind(this),
+          editSession   : this.editSession.bind(this),
+          deleteSession : this.deleteSession.bind(this)
+        }
+      },
+      {
+        name          : 'Speakers',
+        valuePath     : 'speakers',
+        cellComponent : 'ui-table/cell/cell-speakers'
+      },
+      {
+        name            : 'Rating',
+        valuePath       : 'id',
+        extraValuePaths : ['rating', 'feedbacks'],
+        cellComponent   : 'ui-table/cell/events/view/sessions/cell-rating',
+        options         : {
+          ratedSessions: this.ratedSessions
+        },
+        actions: {
+          updateRating : this.updateRating.bind(this),
+          addRating    : this.addRating.bind(this)
+        }
+      },
+      {
+        name            : 'Avg Rating',
+        valuePath       : 'averageRating',
+        isSortable      : true,
+        headerComponent : 'tables/headers/sort'
+      },
+      {
+        name            : 'No. of ratings',
+        valuePath       : 'feedbacks.length',
+        isSortable      : true,
+        headerComponent : 'tables/headers/sort'
+      },
+      {
+        name      : 'Track',
+        valuePath : 'track.name'
+      },
+      {
+        name      : 'Type',
+        valuePath : 'sessionType.name'
```

Comment:
@CosmicCoder96 I tried to display `sessionType.name` and `track.name` using this but at first load it is missing. As soon as I trigger any action such as change tab or rate session or anything else, both of them appears.
Should I add a new cell template file and pass a computed property to it and display this?

Reply:
This action triggering is taking time and other click for other actions too, e.g. changing tab